### PR TITLE
update changelog for v0.3.3 release

### PR DESCRIPTION
**Pull request by asjohnston-asf**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.3.3]
+### Changed
+- Updated GitHub actions to use `provision-with-micromamba` instead of `setup-miniconda`
+- Version upgrades for python dependencies and `ASFHyP3/actions`
```
Comment (jtherrmann) [inline CHANGELOG.md:10]: Do we normally include dependency upgrades in a Changelog entry?

Comment (asjohnston-asf) [reply CHANGELOG.md:10]: No, but that's effectively all this release is, so I felt like calling it out to explain why we did a release

Comment (asjohnston-asf) [reply CHANGELOG.md:10]: No, but that's effectively all this release is, so I felt like calling it out to explain why we did a release

Comment (jtherrmann) [reply CHANGELOG.md:10]: Should we do a bumpless release in that case?

**Review by jtherrmann — APPROVED** [review]

**Pull request merged**